compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp

```diff
@@ -165,6 +176,24 @@ static LogicalResult canReorderWorkgroups(FunctionOpInterface funcOp) {
   return success(workgroupCounts.size() >= 2);
 }
 
+// Returns success when shared memory bank conflict reduction is supported /
+// enabled for `funcOp`.
+static LogicalResult
+canReduceSharedMemoryBankConflicts(FunctionOpInterface funcOp) {
```

Comment:
Would that also work with passes that require more options than just on/off? E.g., max shared mem padding size.

Reply:
Oh, I see what you mean after reading through @hanhanW's snippets. Yeah, that looks like a better way to handle this.

Reply:
Looks like you are only using this logic within the pass that this function is being passed to. Make it local to that?

Reply:
Done. I could not move `canReorderWorkgroups` fully to lower target executable because it needs to know if workgroup counts are static or not, and this happens after tile and distribute.